lib/model/store.dart

```diff
@@ -723,6 +783,7 @@ class UpdateMachine {
     );
     final updateMachine = UpdateMachine.fromInitialSnapshot(
       store: store, initialSnapshot: initialSnapshot);
+    store.updateMachine = updateMachine;
```

Comment:
Let's put this assignment inside the `UpdateMachine` constructor. That way it's more crisply an invariant that the UpdateMachine's store will always point back to it.

Then relatedly (also about making invariants more explicit), `PerAccountStore.updateMachine` can get a setter that asserts the old value was null, so we're not redirecting the same store to start pointing to a different update machine.